PR: https://github.com/odoo/odoo/pull/76011

From: 28836f48a46c0a8a3de711b49f601071f64b33b3
From: Thibault Libioulle
Date: 2021-09-06 14:36:08

Structural Changes: 6
Total Changes: 7

[MOV] project: move the dependent_ids fields to community

This commit moves the project.task dependent_ids field to the community
project module in order to take advantage of this relation for further
development.

closes odoo/odoo#76011

Related: odoo/enterprise#20623
Signed-off-by: LTU-Odoo <IT-Ideas@users.noreply.github.com>

================================= pseudo patch: =================================

--- a/addons/project/models/project.py
+++ b/addons/project/models/project.py
@@ -1013,7 +1013,12 @@ class Task(models.Model):
     # Task Dependencies fields
     allow_task_dependencies = fields.Boolean(related='project_id.allow_task_dependencies')
     # Tracking of this field is done in the write function
-    depend_on_ids = fields.Many2many('project.task', relation="task_dependencies_rel", column1="task_id", column2="depends_on_id", string="Blocked By", domain="[('allow_task_dependencies', '=', True), ('id', '!=', id)]")
+    depend_on_ids = fields.Many2many('project.task', relation="task_dependencies_rel", column1="task_id",
+                                     column2="depends_on_id", string="Blocked By",
+                                     domain="[('allow_task_dependencies', '=', True), ('id', '!=', id)]")
+    dependent_ids = fields.Many2many('project.task', relation="task_dependencies_rel", column1="depends_on_id",
+                                     column2="task_id", string="Block",
+                                     domain="[('allow_task_dependencies', '=', True), ('id', '!=', id)]")
 
     # recurrence fields
     allow_recurring_tasks = fields.Boolean(related='project_id.allow_recurring_tasks')
